### PR TITLE
Sde display expansion boxes

### PR DESCRIPTION
**Pull request by lukoktonos**
This PR adds the UI elements to show results for define statements.
image
image
image

Pull requests into Bonnie require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] This PR is into the correct branch.
- [x] JIRA ticket for this PR: https://jira.mitre.org/browse/BONNIE-1449
- [x] JIRA ticket links to this PR
- [x] Code diff has been done and been reviewed (it **does not** contain: additional white space, not applicable code changes, debug statements, etc.)
- [x] If UI changes have been made, google WAVE plug-in has been executed to ensure no 508 issues were introduced.
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass (remember to update Gemfile when applicable)
- [x] Test fixtures updated and documented as necessary ( see [internal wiki](https://gitlab.mitre.org/bonnie/internal-documentation/wikis/testing#test-fixtures) )
- [x] Code coverage has not gone down and all code touched or added is covered. 

Branch | Back End Coverage | Front End Coverage
-- | -- | --
bonnie-v2.2 | 1738 / 1950 LOC (89.13%) covered | Statements   : 58.94% ( 7134/12104 ) <br>Branches     : 42.79% ( 1935/4522 ) <br>Functions    : 49.19% ( 1178/2395 ) <br>Lines        : 56.81% ( 6408/11279 )
sde_display_expansion_boxes | 1738 / 1950 LOC (89.13%) covered. | Statements   : 59.39% ( 7271/12242 )<br>Branches     : 43.06% ( 1959/4549 )<br>Functions    : 49.79% ( 1207/2424 )<br>Lines        : 57.26% ( 6527/11399 )

- [x] Automated regression test(s) pass

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA test links: N/A
- [x] Justification for using JIRA tests: N/A
- [x] JIRA tests have been added to sprint N/A

**Reviewer 1:**

Name: @hossenlopp
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests: N/A
- [x] JIRA tests have been run and pass N/A
- [x] You agree with the justification for use of JIRA tests or have provided input on why you disagree N/A

**Reviewer 2:**

Name:
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA tests have been run and pass
- [x] You agree with the justification for use of JIRA tests or have provided input on why you disagree

### CODE REVIEW

**Review by hossenlopp — CHANGES_REQUESTED** [review]
These should not show up for functions. I think there should be more space after the button before the next define.

**Comment by hossenlopp** [comment]
Quantites still look odd with the `:` between the value and the unit.

File: app/assets/javascripts/helpers/cql_results_helpers.js.coffee [hunk 1]
```diff
@@ -213,30 +213,56 @@ class CQLResultsHelpers
...
+        keyIndent = key.length + 3
+        prettyResult = prettyResult.concat("#{baseIndentation}#{currentIndentation}#{key}: #{@prettyResult(value, nextIndentLevel, keyIndent)}")
+
+        # append commas if it isn't the last key
+        if key == Object.keys(result)[Object.keys(result).length - 1]
+          prettyResult += '\n'
+        else
+          prettyResult += ',\n'
+
+      if indentLevel >= 3
```
Comment (jbradl11) [inline app/assets/javascripts/helpers/cql_results_helpers.js.coffee]: What case is this `if` block covering?

**Comment by jbradl11** [comment]
The horizontal scroll bars have changed on the patient builder page. I think this is a side effect of css changes. 

image

File: app/assets/javascripts/helpers/cql_results_helpers.js.coffee (hunk 1 above)
Comment (lukoktonos) [reply app/assets/javascripts/helpers/cql_results_helpers.js.coffee]: Good question.  Originally, this was needed to keep the final `}` from having any indentation.  It seems to no longer be needed with the current state of this function.  Will remove and just leave `prettyResult = prettyResult + "#{currentIndentation}}"`

**Review by hossenlopp — APPROVED** [review]

**Review by jbradl11 — APPROVED** [review]
Changes look good. Cool feature.

**Pull request merged**